### PR TITLE
CURA-6627_Store_extra_data_project_files

### PR DESCRIPTION
**Pull request by ninovanhooff**
Created on Behalf of Nallath

https://github.com/Ultimaker/Uranium/pull/567

### CODE REVIEW

**Comment by ninovanhooff** [comment]
Note CI checks failing because the linked uranium branch is outdated.

**Comment by nallath** [comment]
> Looks good on the surface, but I feel unit-tests and a developer documentation are required before we can consider this done.

I've added some extra how to use info about preferences & workspace metadata on https://github.com/Ultimaker/Uranium/wiki/Storing-aditional-data-from-a-plugin

**Comment by ninovanhooff** [comment]
Great documentation. Please add some info on the required CuraApi / Cura for these features.

Can cannot approve this PR because apparently I made it my self. Please remember to create your own PRs

File: plugins/3MFReader/ThreeMFWorkspaceReader.py [hunk 1]
```diff
@@ -732,7 +733,25 @@ def read(self, file_name):
...
+        archive = zipfile.ZipFile(file_name, "r")
+
+        metadata_files = [name for name in archive.namelist() if name.endswith("plugin_metadata.json")]
+
+        result = dict()
+
+        for metadata_file in metadata_files:
+            try:
+                plugin_id = metadata_file.split("/")[0]
+                result[plugin_id] = json.loads(archive.open("Cura/plugin_metadata.json").read().decode("utf-8"))
```
Comment (bradymadams) [inline plugins/3MFReader/ThreeMFWorkspaceReader.py]: @nallath should this be opening metadata_file, rather than "Cura/plugin_metadata.json"?

Comment (rburema) [reply plugins/3MFReader/ThreeMFWorkspaceReader.py]: Seems this was fixed in commit 87fb0d7

**Pull request merged**